src/pyconverter/xml2py/writer.py

```diff
@@ -238,25 +244,107 @@ def copy_package(template_path, new_package_path, clean=False, include_hidden=Fa
 
 
+def write_global__init__file(library_path, module_name_list):
+    """
+    Write the __init__.py file for the package generated.
+
+    Parameters
+    ----------
+    library_path : str
+        Path to the directory containing the generated package.
+
+    name_map : dict
+        Dictionary with the following format: ``{"initial_command_name": "python_name"}``.
+
+    command_map : dict
+        Dictionary with the following format: ``{"initial_command_name": command_object}``.
+
+    structure_map : dict, optional
+        Dictionary with the following format:
+        ``{'module_name': [{'class_name': python_names_list}]}``.
+        The default value is ``None``.
+    """
+    mod_file = os.path.join(library_path, "__init__.py")
+    # TODO: needs to be modified due to the new structure
+
+    with open(mod_file, "w") as fid:
+        fid.write(f"from . import (\n")
+        for dir in os.listdir(library_path):
+            if os.path.isdir(os.path.join(library_path, dir)):
+                fid.write(f"    {dir},\n")
+        fid.write(")\n\n")
+        fid.write("try:\n")
+        fid.write("    import importlib.metadata as importlib_metadata\n")
+        fid.write("except ModuleNotFoundError:\n")
+        fid.write("    import importlib_metadata\n\n")
+        fid.write("__version__ = importlib_metadata.version(__name__.replace('.', '-'))\n")
+        fid.write('"""PyConverter-GeneratedCommands version."""\n')
+    fid.close()
+
+
+def write__init__file(library_path):
+
+    for dir in os.listdir(library_path):
+        if os.path.isdir(os.path.join(library_path, dir)):
+            listdir = os.listdir(os.path.join(library_path, dir))
+            if len(listdir) > 0:
+                with open(os.path.join(library_path, dir, "__init__.py"), "w") as fid:
+                    fid.write(f"from . import (\n")
+                    for file in listdir:
+                        if file.endswith(".py"):
+                            fid.write(f"    {file[:-3]},\n")
+                    fid.write(")\n")
+                    fid.close()
+
+
+def get_library_path(new_package_path, config_path):
+    """
+    Return the desired library path with the following format:
+    ``new_package_path/library_structure``.
+
+    For instance, if ``library_name_structured`` in the ``config.yaml`` file is
+    ``["pyconverter", "generatedcommands"]``, the function will return
+    ``new_package_path/pyconverter/generatedcommands``.
+
+    Parameters
+    ----------
+    new_package_path : str
+        Path to the new package directory.
+    config_path : str
+        Path to the configuration file.
+
+    Returns
+    -------
+    str
+        Path to the desired library structure.
+    """
+    library_name = get_config_data_value(config_path, "library_name_structured")
+    if not "src" in library_name:
+        library_name.insert(0, "src")
+    library_name_str = "/".join(library_name)
+    return os.path.join(new_package_path, library_name_str)
```

Comment:
[os.path.join](https://docs.python.org/3/library/os.path.html#os.path.join) accepts multiple arguments:
```suggestion
    return os.path.join(new_package_path, *library_name)
```

Reply:
I couldn't remember how to do it, thanks for the tip!